### DISCUSSION TITLE
Should `requestCrawl` be a "procedure" (POST) rather than a "query" (GET)?

### DISCUSSION THREAD

**Discussion by DavidBuchanan314** [bluesky-social/atproto#1214] [General Discussion]
HTTP GET requests are supposed to be idempotent, but I'm not sure the behavior of `requestCrawl` strictly matches that.

**Comment by bnewbold** [comment]
I think you are probably correct, and this should be a "procedure" not "query" endpoint, mapping to an HTTP POST request.

Curious what @dholms thinks.

**Reply by dholms** [reply]
Yup you're right 👍 

Tbh not sure what was going through my head when I sketched out that schema a few months back

**Comment by whyrusleeping** [comment]
requestCrawl *is* idempotent though, multiple calls with the same params should result in a no-op

**Comment by snarfed** [comment]
...but it has side effects, which ideally `GET` shouldn't, right? Also it triggers a relatively expensive/heavyweight operation, so `POST` might be more prudent in practical terms, regardless of which method's theoretical/standards semantics it matches best?

**Comment by whyrusleeping** [comment]
Yeah, it should be a post regardless, agreed

**Comment by sugyan** [comment]
One suggestion related to this discussion, how about removing `optional()` from `output` in `query`?
https://github.com/bluesky-social/atproto/blob/main/packages/lexicon/src/types.ts#L231
There should be no method that has neither side-effects nor return value.

I thought this would prevent the confusion between `query` and `procedure` definitions, as in this case.
